spec/features/parent_gives_consent_spec.rb

```diff
@@ -0,0 +1,350 @@
+require "rails_helper"
+
+module SessionCreationSteps
```

Comment:
This module is an experiment. It's currently only used in this spec but I suspect it's going to be reusable beyond this spec, and so could be extracted to a helper file. It could also help maintenance – if microcopy or journey tweaks are made, we don't want lots of feature specs breaking.

Reply:
Just a couple thoughts ...

- I think I much prefer it when the top of the file is the journey being tested, so I do think if we keep this Steps module we should put it in a separate file
- I generally don't mind having a file with common steps, but I think it should be used sparingly: while it was a slight pain with the Playwright tests to update multiple tests for a small front-end change, I think it was better than having to remember all the available step methods when building a test. I'd probably err on the side of less shared steps

Reply:
> I think I much prefer it when the top of the file is the journey being tested, so I do think if we keep this Steps module we should put it in a separate file

yes, agreed. I've added that in [2a37357](https://github.com/nhsuk/manage-childrens-vaccinations/pull/1049/commits/2a373573ebc789dca47945b719c55dc3ceb22955)

> I generally don't mind having a file with common steps, but I think it should be used sparingly: while it was a slight pain with the Playwright tests to update multiple tests for a small front-end change, I think it was better than having to remember all the available step methods when building a test. I'd probably err on the side of less shared steps

Yep, have definitely been in that "pile of steps" scenario on previous projects and it's another type of maintenance nightmare.

I think there's a balance that could be struck if we don't go overboard with it; I think there's a fair amount of value in being able to tersely define a journey that maybe isn't core to what's being specced in that particular scenario – kinda like how you can skip a lot of the details when using FactoryBot factories if you don't care about them. Here's an example of what would be nice to do:

```ruby
  def and_the_sais_team_has_organised_a_session_at_my_childs_school
    sign_in_as_nurse_testy
    start_new_session(
      school_name: @school.name,
      session_date: 1.week.from_now,
      time_of_day: "Morning"
    )
    select_cohort(names: [@child.full_name])
    schedule_key_dates(send_consent_on_date: Time.zone.today)
    confirm_session_details(
      number_of_children: 1,
      send_consent_on_date: Time.zone.today,
      session_date: 1.week.from_now
    )
  end
```

I guess we could rely on factories to achieve the same thing, but having fixed a few factory bugs recently, it's nice to be able to do stuff just through the frontend.

Reply:
Updated here [2a37357](https://github.com/nhsuk/manage-childrens-vaccinations/pull/1049/commits/2a373573ebc789dca47945b719c55dc3ceb22955)

Reply:
@benilovj I think you've got the right idea 👍 

**Do:** abstract reusable setup and repetitive code that gets used **inside** steps

```ruby
# spec/system/foo_spec.rb
def when_i_am_signed_in
  sign_in_as_nurse_testy
end

# spec/system/bar_spec.rb
def and_i_am_signed_in
  sign_in_as_nurse_testy
end

# spec/system/baz_spec.rb
def then_i_sign_in
  sign_in_as_nurse_testy
end
```

**Don't**: abstract the **actual** steps, those are more like documentation and should be more flexible

```ruby
# DONT:
include SharedSteps

# ...
when_i_am_signed_in
```